include/aikido/control/BarrettHandKinematicSimulationPositionCommandExecutor.hpp

```diff
@@ -0,0 +1,95 @@
+#ifndef AIKIDO_CONTROL_BARRETTHANDKINEMATICSIMULATIONPOSITIONCOMMANDEXECUTOR_HPP_
+#define AIKIDO_CONTROL_BARRETTHANDKINEMATICSIMULATIONPOSITIONCOMMANDEXECUTOR_HPP_
+#include <aikido/control/PositionCommandExecutor.hpp>
+#include <aikido/control/BarrettFingerKinematicSimulationPositionCommandExecutor.hpp>
+#include <aikido/control/BarrettFingerKinematicSimulationSpreadCommandExecutor.hpp>
+#include <dart/collision/CollisionDetector.hpp>
+#include <dart/collision/CollisionOption.hpp>
+#include <dart/collision/CollisionGroup.hpp>
+#include <dart/collision/CollisionFilter.hpp>
+#include <dart/dynamics/dynamics.hpp>
+#include <Eigen/Dense>
+#include <future>
+#include <mutex>
+#include <chrono>
+
+namespace aikido {
+namespace control {
+
+/// Position command executor for simulating BarrettHand fingers.
+/// Assumes that fingers are underactuated: proximal joint is actuated
+/// and distal joint moves with certain mimic ratio until collision.
+class BarrettHandKinematicSimulationPositionCommandExecutor
+: public PositionCommandExecutor
+{
+public:
+  /// Constructor.
+  /// \param[in] positionCommandExecutors 3 executors to control
+  ///        proximal and distal joints of the fingers. The first two fingers
+  ///        should have (spread, proximal, distal) joints.
+  ///        Third finger should have 2 joints (proximal, distal).
+  /// \param[in] spreadCommandExecutor Executors to control
+  ///        spreads of the fingers.
+  /// \param[in] collideWith CollisionGroup to check collision with fingers.
+  BarrettHandKinematicSimulationPositionCommandExecutor(
+    const std::array<
+      BarrettFingerKinematicSimulationPositionCommandExecutorPtr, 3>& positionCommandExecutors,
+    BarrettFingerKinematicSimulationSpreadCommandExecutorPtr spreadCommandExecutor,
+    ::dart::collision::CollisionGroupPtr collideWith);
```

Comment:
I wound up basically copying and pasting the optional nullptr logic from `BarrettFingerKinematicSimulationPositionCommandExecutor`. Since the two classes are completely independent I didn't see an obvious way to avoid that... :cry: